From 1dc598c3a13e5418488561425f056e1120d273f5 Mon Sep 17 00:00:00 2001
From: Yang Shi <yang.shi@windriver.com>
Date: Tue, 10 Sep 2013 18:54:40 -0700
Subject: [PATCH] Samples: Clean up so modules and hostprogs build correctly

Add the samples directory to the core so that modules will
build.

Only build hostprogs if not cross compiling.  Otherwise the
resulting executables would be pretty much useless.

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
[Uprev to apply to 3.10]
Signed-off-by: Yang Shi <yang.shi@windriver.com>

diff --git a/Makefile b/Makefile
index 9d01c62c94cc..573f756f987f 100644
--- a/Makefile
+++ b/Makefile
@@ -797,7 +797,7 @@ export mod_sign_cmd
 
 
 ifeq ($(KBUILD_EXTMOD),)
-core-y		+= kernel/ mm/ fs/ ipc/ security/ crypto/ block/ grsecurity/
+core-y		+= kernel/ mm/ fs/ ipc/ security/ crypto/ block/ grsecurity/ samples/
 
 vmlinux-dirs	:= $(patsubst %/,%,$(filter %/, $(init-y) $(init-m) \
 		     $(core-y) $(core-m) $(drivers-y) $(drivers-m) \
@@ -836,9 +836,6 @@ vmlinux: scripts/link-vmlinux.sh $(vmlinux-deps) FORCE
 ifdef CONFIG_HEADERS_CHECK
 	$(Q)$(MAKE) -f $(srctree)/Makefile headers_check
 endif
-ifdef CONFIG_SAMPLES
-	$(Q)$(MAKE) $(build)=samples
-endif
 ifdef CONFIG_BUILD_DOCSRC
 	$(Q)$(MAKE) $(build)=Documentation
 endif
diff --git a/samples/hidraw/Makefile b/samples/hidraw/Makefile
index 1e6aabc009f1..583afd5d7f25 100644
--- a/samples/hidraw/Makefile
+++ b/samples/hidraw/Makefile
@@ -4,7 +4,9 @@ obj- := dummy.o
 # List of programs to build
 hostprogs-y := hid-example
 
+ifeq ($(CROSS_COMPILE),)
 # Tell kbuild to always build the programs
 always := $(hostprogs-y)
+endif
 
 HOSTCFLAGS_hid-example.o += -I$(srctree)/usr/include
diff --git a/samples/seccomp/Makefile b/samples/seccomp/Makefile
index 64648e4632c6..cf810c3a81d7 100644
--- a/samples/seccomp/Makefile
+++ b/samples/seccomp/Makefile
@@ -38,5 +38,7 @@ HOSTLOADLIBES_dropper += $(MFLAG)
 endif
 endif
 
+ifeq ($(CROSS_COMPILE),)
 # Tell kbuild to always build the programs
 always := $(hostprogs-y)
+endif
-- 
1.8.5.2

